pkg/webhooks/policyvalidation.go

```diff
@@ -31,10 +31,25 @@ func (ws *WebhookServer) handlePolicyValidation(request *v1beta1.AdmissionReques
 	// check for uniqueness of rule names while CREATE/DELET
 	admissionResp = ws.validateUniqueRuleName(policy)
 
-	if admissionResp.Allowed {
-		ws.manageWebhookConfigurations(*policy, request.Operation)
+	// helper function to evaluate if policy has validtion or mutation rules defined
+	hasMutateOrValidate := func() bool {
+		for _, rule := range policy.Spec.Rules {
+			if !reflect.DeepEqual(rule.Mutation, kyverno.Mutation{}) || !reflect.DeepEqual(rule.Validation, kyverno.Validation{}) {
+				glog.Infoln(rule.Name)
+				return true
+			}
+		}
+		return false
 	}
 
+	if admissionResp.Allowed {
+		if hasMutateOrValidate() {
```

Comment:
Do we handle delete action of policy to remove webhooks here?

Reply:
the webhookconfigurations do no observer delete operation

Reply:
the resource webhook deletion is handled in the policy controller

Reply:
Did we remove this from webhookconfigurations? I'm thinking we may want to prevent deletion of webhookconfigurations through webhook.

Reply:
so we used to watch for DELETE operations on the resource, with the old policyViolation design where we removed the policy violation from policy status. But after using CRD and controller redesign.